sanic/app.py

```diff
@@ -1027,8 +1027,15 @@ def run(
                 "stop_event will be removed from future versions.",
                 DeprecationWarning,
             )
-        # compatibility old access_log params
-        self.config.ACCESS_LOG = access_log
+        # if access_log is passed explicitly change config.ACCESS_LOG
+        if access_log is not None:
+            if isinstance(access_log, bool):
```

Comment:
I don't think this is a good pattern for type checking. If we assume users can input any types, then other params will have the same issue, and we do the same thing for all params ?

IMO, we should just put all param types in docstring and shows them in documentation.